api/src/main/java/org/pmiops/workbench/api/ConceptSetsController.java

```diff
@@ -358,6 +360,63 @@ private void addConceptsToSet(
     return ResponseEntity.ok(toClientConceptSet(dbConceptSet));
   }
 
+  @Override
+  public ResponseEntity<ConceptSet> copyConceptSet(
+      String fromWorkspaceNamespace,
+      String fromWorkspaceId,
+      String fromConceptSetId,
+      CopyRequest copyRequest) {
+    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
+    workspaceService.enforceWorkspaceAccessLevel(
+        fromWorkspaceNamespace, fromWorkspaceId, WorkspaceAccessLevel.READER);
+    Workspace toWorkspace =
+        workspaceService.get(
+            copyRequest.getToWorkspaceNamespace(), copyRequest.getToWorkspaceName());
+    Workspace fromWorkspace = workspaceService.get(fromWorkspaceNamespace, fromWorkspaceId);
+    workspaceService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
+        toWorkspace.getWorkspaceNamespace(),
+        toWorkspace.getFirecloudName(),
+        WorkspaceAccessLevel.WRITER);
+    if (toWorkspace.getCdrVersion().getCdrVersionId()
+        != fromWorkspace.getCdrVersion().getCdrVersionId()) {
+      throw new BadRequestException(
+          "New workspace does not have the same CDR version as current workspace");
+    }
+    org.pmiops.workbench.db.model.ConceptSet conceptSet =
```

Comment:
what if it can't be found? Does it throw? It's possible the concept set could have been deleted or had its access level changed just before we get here.

Reply:
I believe this would throw an error. I will verify.

Reply:
It will return null. I am handling now.